src/docfx/lib/git/GitUtility.cs

```diff
@@ -118,6 +118,46 @@ public static Task Reset(string cwd, string branch)
         public static Task<string> HeadRevision(string cwd)
            => ExecuteQuery(cwd, "rev-parse HEAD", TimeSpan.FromMinutes(3));
 
+        /// <summary>
+        /// Retrieve original URL
+        /// </summary>
+        public static Task<string> GetOriginalUrl(string cwd)
```

Comment:
For simple queries like this, *libgit* or *libgit2sharp* is way faster than *launching git exe*. *git exe* is way faster in IO bound operations like *clone*, *pull*, *checkout*

Reply:
add a TODO

---
In reply to: [194685951](https://github.com/dotnet/docfx/pull/2876#discussion_r194685951) [](ancestors = 194685951)